data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -102,9 +104,19 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
 
         includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
         excludeKeysSet.addAll(keyValueProcessorConfig.getExcludeKeys());
+        defaultKeysMap.putAll(keyValueProcessorConfig.getDefaultKeys());
+        if (!defaultKeysMap.isEmpty()) {
+            defaultKeysSet.addAll(defaultKeysMap.keySet());
+        }
 
-        validateKeySets(includeKeysSet, excludeKeysSet);
+        validateKeySets(includeKeysSet, excludeKeysSet, defaultKeysSet);
 
+        final Set<String> includeDefaultCheckSet = new HashSet<String>(includeKeysSet);
+        includeDefaultCheckSet.retainAll(defaultKeysSet);
+        if (!includeDefaultCheckSet.isEmpty()) {
+            includeKeysSet.removeAll(defaultKeysSet);
+        }
```

Comment:
Why do we remove default keys from includeKeys? e.g.
include_keys = ["key1"]
defaultKeysSet = ["key1", "dkey"]
shall we expect either of the following to appear in the parsed message?
* both `key1` and `dkey`
* only `key1`

Reply:
I remove default keys from include keys to avoid adding the key twice, since I implemented default key-value pairs to always be added to the parsed message no matter what

Reply:
For your example here, we should have both `key1` and `dkey` included. But I see your comment on the test file, so will change some implementation